AutoPartsShop.API/Controllers/ProductsController.cs

```diff
@@ -0,0 +1,140 @@
+                return StatusCode(StatusCodes.Status500InternalServerError,
+                                 "Error retrieving data from the database");
+            }
+        }
+
+        [HttpGet("{id:int}")]
+        public async Task<ActionResult<ProductDTO>> GetItem(int id)
+        {
+            try
+            {
+                var product = await this.productRepository.GetItem(id);
+
+                if (product == null)
+                {
+                    return BadRequest();
+                }
+                else
+                {
+                    var productCategory = await this.productRepository.GetCategory(product.CategoryId);
+
+                    var productDTO = new ProductDTO
+                    {
+                        Id = product.Id,
+                        ProductName = product.ProductName,
+                        Brand = product.Brand,
+                        Description = product.Description,
+                        ImageURL = product.ImageURL,
+                        Price = product.Price,
+                        Qty = product.Qty,
+                        CategoryId = product.CategoryId,
+                        CategoryName = productCategory?.CategoryName
+                    };
+
+                    return Ok(productDTO);
+                }
+            }
+            catch (Exception)
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError,
+                                 "Error retrieving data from the database");
+            }
+        }
+
+        [HttpGet]
+        [Route(nameof(GetProductCategories))]
+        public async Task<ActionResult<IEnumerable<ProductCategoryDTO>>> GetProductCategories()
+        {
+            try
+            {
+                var productCategories = await productRepository.GetCategories();
+
+                var productCategoriesDTOs = productCategories.ConvertToDTO();
+
+                return Ok(productCategoriesDTOs);
+            }
+            catch (Exception)
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError,
+                                 "Error retrieving data from the database");
+            }
+
+        }
+
+        [HttpGet]
+        [Route("{categoryId}/GetItemsByCategory")]
+
+        public async Task<ActionResult<IEnumerable<ProductDTO>>> GetItemsByCategory(int categoryId)
+        {
+            try
+            {
+                var products = await productRepository.GetItemsByCategory(categoryId);
+                var productCategories = await productRepository.GetCategories();
+                var productDTOs = products.ConvertToDTO(productCategories);
+
+                return Ok(productDTOs);
+            }
+            catch (Exception)
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError,
+                                                 "Error retrieving data from the database");
+            }
+        }
+
+
```

Comment:
I see no delete, post, put or patch methods for the products. That means that there is no way to create, update or remove a product?